### PR TITLE
Improve focus handling from scroller re-rendering all items and losing focus

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 90321277

## Summary

Refactors the focus handling on the RecycleScroller, in instances where it updates the elements, from open/close events or from scrolling the virtual scroller.

## Dependencies

NA

## Testing

Serve the app with a doccarchive that has the navigator data

Steps:
1. Using only the keyboard, start navigating with Tabbing across the items.
2. Once you focus one of the last items at the end of the screen, it will jump to the next one, moving it to the center of the scrollable area
3. Assert that focus is still kept on the item. (we used to loose focus on that element here)
4. Assert that after focusing, initiating a filter (by tag or text) resets the focus order. (try to focus back into the items)
5. Assert that clicking on items and navigating to their page, focus is still kept on it.
6. Assert that after focusing, somewhere inside the content moves the focus away from the navigator.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

**Review by marinaaisa — CHANGES_REQUESTED** [review]
Looks and works quite well, I just added a small comment! Thanks a lot, Dobri, I was running out of ideas with this bug 😄 ✨

File: src/components/Navigator/NavigatorCard.vue [hunk 1]
```diff
@@ -33,9 +33,12 @@
             aria-label="Sidebar Tree Navigator"
             :items="nodesToRender"
             :item-size="itemSize"
+            emit-update
```
[the remark this answers is not in the document]
Comment (dobromir-hristov) [reply src/components/Navigator/NavigatorCard.vue:36]: No, we need this for `@update`

**Comment by dobromir-hristov** [comment]
@swift-ci test

File: src/components/Navigator/NavigatorCard.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (marinaaisa) [reply src/components/Navigator/NavigatorCard.vue:36]: Oh! I see that it's a vue-virtual-scroller's prop! Perfect!

**Review by marinaaisa — APPROVED** [review]

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**